velero-plugin-for-gcp/object_store.go

```diff
@@ -76,12 +77,32 @@ func newObjectStore(logger logrus.FieldLogger) *ObjectStore {
 }
 
 func (o *ObjectStore) Init(config map[string]string) error {
-	if err := veleroplugin.ValidateObjectStoreConfigKeys(config, kmsKeyNameConfigKey, serviceAccountConfig); err != nil {
+	if err := veleroplugin.ValidateObjectStoreConfigKeys(config, kmsKeyNameConfigKey, serviceAccountConfig, credentialsFileConfigKey); err != nil {
 		return err
 	}
 	// Find default token source to extract the GoogleAccessID
 	ctx := context.Background()
-	creds, err := google.FindDefaultCredentials(ctx)
+
+	clientOptions := []option.ClientOption{
+		option.WithScopes(storage.ScopeReadWrite),
+	}
+
+	var creds *google.Credentials
+	var err error
+
+	// Prioritize the credentials file path in config, if it exists
+	if credentialsFile, ok := config[credentialsFileConfigKey]; ok {
+		b, err := ioutil.ReadFile(credentialsFile)
+		if err != nil {
+			return errors.Wrapf(err, "error reading provided credentials file %v", credentialsFile)
+		}
+		creds, err = google.CredentialsFromJSON(ctx, b)
+		clientOptions = append(clientOptions, option.WithCredentialsFile(credentialsFile))
```

Comment:
I'm not sure why we need this line. The variable `creds` had configured at line 99 and https://github.com/vmware-tanzu/velero-plugin-for-gcp/blob/15a3a27e8da0d22a74472c3ffe9039c0a9cb9515/velero-plugin-for-gcp/object_store.go#L111-L113 will use that credential, am I right?

Reply:
Hmm, there was a reason that I needed this. Let me find it in the docs and I'll add a comment. Thanks for pointing this out!

Reply:
@jenting I've looked at this again. There are two separate uses for the credentials file: the first is to create the client to communicate with GCP, and the other is to configure the credentials to create signed URLs. By default if a client option to use a specific credential is not provided, it will default to using the `GOOGLE_APPLICATION_CREDENTIALS` internally. So that's why we didn't explicitly provide that before. I've added comments but let me know if you think it needs more explanation :)